packages/crypto/src/derivation/utils.ts

```diff
@@ -0,0 +1,100 @@
+import { stringToBytes } from "@scure/base"
+
+import { addressEncodingFromCurve, addressFromPublicKey } from "../address"
+import { entropyToSeed, getDevSeed, isValidMnemonic, mnemonicToEntropy } from "../mnemonic"
+import { KeypairCurve } from "../types"
+import { deriveEcdsa, getPublicKeyEcdsa } from "./deriveEcdsa"
+import { deriveEd25519, getPublicKeyEd25519 } from "./deriveEd25519"
+import { deriveEthereum, getPublicKeyEthereum } from "./deriveEthereum"
+import { deriveSolana, getPublicKeySolana } from "./deriveSolana"
+import { deriveSr25519, getPublicKeySr25519 } from "./deriveSr25519"
+
+export const deriveKeypair = (seed: Uint8Array, derivationPath: string, curve: KeypairCurve) => {
+  switch (curve) {
+    case "sr25519":
+      return deriveSr25519(seed, derivationPath)
+    case "ed25519":
+      return deriveEd25519(seed, derivationPath)
+    case "ecdsa":
+      return deriveEcdsa(seed, derivationPath)
+    case "ethereum":
+      return deriveEthereum(seed, derivationPath)
+    case "solana":
+      return deriveSolana(seed, derivationPath)
+  }
+}
+
+export const getPublicKeyFromSecret = (secretKey: Uint8Array, curve: KeypairCurve): Uint8Array => {
+  switch (curve) {
+    case "ecdsa":
+      return getPublicKeyEcdsa(secretKey)
+    case "ethereum":
+      return getPublicKeyEthereum(secretKey)
+    case "sr25519":
+      return getPublicKeySr25519(secretKey)
+    case "ed25519":
+      return getPublicKeyEd25519(secretKey)
+    case "solana":
+      return getPublicKeySolana(secretKey)
+  }
+}
+
+export const addressFromSuri = (suri: string, type: KeypairCurve) => {
+  const { mnemonic, derivationPath, password } = parseSuri(suri)
+
+  const entropy = mnemonicToEntropy(mnemonic)
+  const seed = entropyToSeed(entropy, type, password) // ~80ms
+  const { secretKey } = deriveKeypair(seed, derivationPath, type)
+  const publicKey = getPublicKeyFromSecret(secretKey, type)
+  const encoding = addressEncodingFromCurve(type)
+
+  return addressFromPublicKey(publicKey, encoding)
+}
+
+// @dev: we only expect suri to contain a mnemonic and derivation path
+// for other cases see https://polkadot.js.org/docs/keyring/start/suri/
+export const parseSuri = (suri: string) => {
```

Comment:
Made this comment show up in vscode when using the method in other files!

```suggestion

/**
 * @dev we only expect suri to contain a mnemonic and derivation path.
 * for other cases see https://polkadot.js.org/docs/keyring/start/suri/
 */
export const parseSuri = (suri: string) => {
```